madara/node/src/service/l1.rs

```diff
@@ -107,32 +104,56 @@ where
         sync_config: L1SyncConfig<'_>,
         settlement_client: Option<Arc<dyn SettlementClientTrait<Config = C, StreamType = S>>>,
     ) -> anyhow::Result<Self> {
-        let gas_price_sync_enabled = sync_config.authority
-            && !sync_config.devnet
-            && (config.gas_price.is_none() || config.blob_gas_price.is_none());
-        let gas_price_poll = config.gas_price_poll;
-
-        if gas_price_sync_enabled {
-            let settlement_client =
-                settlement_client.clone().context("L1 gas prices require the service to be enabled...")?;
-            tracing::info!("⏳ Getting initial L1 gas prices");
-            mc_settlement_client::gas_price::gas_price_worker_once(
-                settlement_client,
-                &sync_config.l1_gas_provider,
-                gas_price_poll,
-                sync_config.l1_block_metrics.clone(),
+        let gas_price_needed = sync_config.authority && !sync_config.devnet;
+        let gas_provider_config = if gas_price_needed {
+            let mut gas_price_provider_builder = GasPriceProviderConfigBuilder::default();
+            if let Some(fix_gas) = config.gas_price {
+                gas_price_provider_builder.set_fix_gas_price(fix_gas);
+            }
+            if let Some(fix_blob_gas) = config.blob_gas_price {
+                gas_price_provider_builder.set_fix_data_gas_price(fix_blob_gas);
+            }
+            if let Some(strk_per_eth_fix) = config.strk_per_eth {
+                gas_price_provider_builder.set_fix_strk_per_eth(strk_per_eth_fix);
+            }
+            if let Some(ref oracle_url) = config.oracle_url {
+                if let Some(ref oracle_api_key) = config.oracle_api_key {
+                    let oracle = PragmaOracleBuilder::new()
+                        .with_api_url(oracle_url.clone())
+                        .with_api_key(oracle_api_key.clone())
+                        .build();
+                    gas_price_provider_builder.set_oracle_provider(Arc::new(oracle));
+                } else {
+                    bail!("Only Pragma oracle is supported, please provide the oracle API key");
+                }
+            }
+            Some(
+                gas_price_provider_builder
+                    .with_poll_interval(config.gas_price_poll)
+                    .build()
+                    .context("Building gas price provider config")?,
             )
-            .await
-            .context("Getting initial gas prices")?;
+        } else {
+            None
+        };
+
+        if let Some(config) = gas_provider_config.as_ref() {
+            if !config.all_is_fixed() {
+                let settlement_client =
+                    settlement_client.clone().context("L1 gas prices require the service to be enabled...")?;
+                tracing::info!("⏳ Getting initial L1 gas prices");
+                let l1_gas_quote = mc_settlement_client::gas_price::update_gas_price(settlement_client, config)
```

Comment:
cool, i think we should add another comment here telling why this exists
(this exists because we want to be 100% sure we already have gas prices fetched before launching block prod and producing a header)